components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerV2.java

```diff
@@ -1125,7 +1125,13 @@ private void doUpdateGroups(String roleId, Set<String> newGroupIDList, Set<Strin
                         new ArrayList<>(deleteGroupIDList), tenantDomain);
             } catch (IdentityRoleManagementException e) {
                 if (RoleConstants.Error.INVALID_REQUEST.getCode().equals(e.getErrorCode())) {
-                    throw new BadRequestException(e.getMessage());
+                    // Error message and SCIM type
+                    String errorMessage = "Updating groups of the role by display name is not supported. "
```

Comment:
Is this the correct place to handle the mentioned error?
Issue should be directly taking `display eq ...` something as group id 
We have to throw the error from that place

Reply:
Addressed in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/557/commits/b974ba9606102966ce3ceb702105778fa9623348